haystack/nodes/prompt/prompt_node.py

```diff
@@ -443,6 +444,12 @@ def __init__(
             if key in kwargs
         }
 
+    def resolve_url(self) -> str:
+        return "https://api.openai.com/v1/completions"
+
+    def resolve_headers(self) -> Dict[str, str]:
+        return {"Authorization": f"Bearer {self.api_key}", "Content-Type": "application/json"}
```

Comment:
I don't think this is necessary to have as separate class functions right now since there is no logic being handled here and nothing is being "resolved". So I think to help with readability I would keep the code as it was before.

Reply:
I made them properties, that should be an ok solution and we keep the current code for invocation, no "resolve" used